lib/utils/voip/permission_service.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:io';
+import 'package:permission_handler/permission_handler.dart';
+
+class PermissionHandlerService {
+  static final PermissionHandlerService _instance = PermissionHandlerService._internal();
+
+  factory PermissionHandlerService() {
+    return _instance;
+  }
+
+  PermissionHandlerService._internal();
+
+
+  Future<PermissionStatus>? requestPermissionForPhotoActions() {
+    if (Platform.isIOS) {
+      return _handlePhotosPermissionIOSAction();
+    } else if (Platform.isAndroid) {
+      return _handlePhotosPermissionAndroidAction();
+    } else {
+      return null;
+    }
+  }
+
+  Future<PermissionStatus> _handlePhotosPermissionIOSAction() async {
+    final currentStatus = await Permission.photos.status;
+    return _handlePermission(currentStatus);
+  }
+
+  Future<PermissionStatus> _handlePhotosPermissionAndroidAction() async {
+    final currentStatus = await Permission.storage.status;
+    return _handlePermission(currentStatus);
+  }
+
+
+  Future<PermissionStatus> _handlePermission(PermissionStatus currentStatus) async {
```

Comment:
`_handlePermission`=> `_handlePhotoPermission` because maybe we have a lot of other permissions